src/wallet/coinselection.cpp

```diff
@@ -459,7 +459,7 @@ CAmount SelectionResult::GetSelectionWaste(CAmount change_cost, CAmount target,
     CAmount selected_effective_value = 0;
     for (const auto& coin_ptr : m_selected_inputs) {
         const COutput& coin = *coin_ptr;
-        waste += coin.GetFee() - coin.long_term_fee;
+        waste += coin.GetFee() + coin.ancestor_bump_fees - coin.long_term_fee;
```

Comment:
nit: no need to add it here, only to remove in the next commit

Reply:
If I don’t add it here, I don’t think the tests would pass.

Reply:
I pulled the part of the waste test that did not deal with the group discount into this commit and actually realized this line was incorrect (and then correctly removed in the next commit). Thank you for pointing this out.